src/App.js

```diff
@@ -1,10 +1,122 @@
 import React from 'react';
+import Header from './components/Header';
+import Main from './components/Main';
+import Footer from './components/Footer';
 
-function App() {
-  return (
-    <div className="App">
-    </div>
-  );
-}
+export default class App extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            todoData: [],
+            filter: 'all'
+        };
+
+    }
+    flag = true;
+    counter = 1;
+
+    createTodoItem(text) {
+        return {
+            text,
+            done: false,
+            id: this.counter++
+        }
+    }
 
-export default App;
+    addItem = (newTodo) => {
+        if (newTodo === '') {
+            return;
+        }
+        this.setState(({todoData}) => {
+                let arrTodoData = [...todoData, this.createTodoItem(newTodo)];
+                return {todoData: arrTodoData}
+            }
+        )
+    };
+    findElement = (todoData, id) => {
+        return todoData.findIndex((el) => el.id === id);
+    };
+
+    onChecked = (id) => {
+        this.setState(({todoData}) => {
+            const copy = [...todoData];
+            const index = this.findElement(copy, id);
+            let obj = copy[index];
+            obj.done = !obj.done;
+            return {
+                todoData: copy
+            }
+        })
+    };
+    checkAll = () => {
+        this.setState(({todoData}) => {
+            const elements = [...todoData];
+            elements.forEach(el => {
+                el.done = this.flag;
```

Comment:
The same. Even if you copy array, the items in array are objects. So you copy links to that objects. Be careful with mutations.